File: api_nestjs_distributed_order_platform/src/application/use-case/order/get-order.use-case.ts
import { Inject, Injectable, NotFoundException } from "@nestjs/common";
import type { OrderRepositoryInterface } from "src/application/port/order.repository";
import { OrderEntity } from "src/domain/entities/order.entity";
import { ORDER_REPOSITORY } from "src/shared/tokens_nest/order.token";

@Injectable()
export class GetOrderUseCase {
    constructor(
        @Inject(ORDER_REPOSITORY)
        private readonly orderRepository: OrderRepositoryInterface,
    ) {}

    async execute(orderId: string): Promise<OrderEntity> {
        const order = await this.orderRepository.findById(orderId);
        if (!order) {
            throw new NotFoundException('Order not found');
        }
        return order;
    }
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/application/usecase/order/GetOrderUseCase.java
package com.project.order.application.usecase.order;

import com.project.order.application.exception.NotFoundException;
import com.project.order.application.port.OrderRepository;
import com.project.order.domain.model.Order;
import org.springframework.stereotype.Service;

@Service
public class GetOrderUseCase {

    private final OrderRepository orderRepository;

    public GetOrderUseCase(OrderRepository orderRepository) {
        this.orderRepository = orderRepository;
    }

    public Order execute(String orderId) {
        return orderRepository.findById(orderId)
                .orElseThrow(() -> new NotFoundException("Order not found"));
    }
}
